### PR TITLE
Added Silo and PB Cannon buildings. Added superweapons.

### PR DESCRIPTION
**Pull request by Dzierzan**
Things to be done:
- Nuclear rocket needs a thruster tail which has to work as an overlay sprite for the projectile.
- ~~`SupportPowers` wants `chrome` palette for clock. I added dummy palette to avoid the crash.~~ EDIT: Frankly, it can be null.
- `SupportPowers` needs to be able to support faction suffix for clock.
- `NukePower` needs to be able to support faction suffix for icons.
- Custom trait for playing silo opening sound (in `INotifySupportPower.Activated` callback)

Notes:
- Damage - nuclear rocket can destroy Shelter and it has 2000 HP. So I concluded the least amount of damage it deals is 2000.
- According to the HD version, it takes 4 mins and 20 seconds to reload.
- The weapon spread damage is 4 tiles.
- Added missing damage state sprites.
- Cut off feature which elevators use, needs to be added for silo in the future (pure cosmetic thing).
- The silo starts fully charged as it is in vanilla.
- Multiple silos are allowed (no hard-cap, though)

silo

### CODE REVIEW

**Comment by Dzierzan** [comment]
`WithSupportPowerActivationAnimation` could support playing sound as both silo and plasmablast have sort of opening sound. PlasmaBlast has a delay between playing animation and actually firing.

Right now opening sound was moved to weapon which is IMO bad in the long run.

**Comment by michaeldgg2** [comment]
> PlasmaBlast has a delay between playing animation and actually firing.

`NukePower` has `MissileDelay`:
```
[Desc("Delay (in ticks) after launch until the missile is spawned.")]
public readonly int MissileDelay = 0;
```
Can't you use this?

> WithSupportPowerActivationAnimation could support playing sound as both silo and plasmablast have sort of opening sound. 

Support powers in general support `LaunchSound` and `IncomingSound`. `LaunchSound` is played when the support power is yours or one of your allies and `IncomingSound` if it belongs to an enemy.

It is `UI`-type sound, but can it be used?

**Comment by Dzierzan** [comment]
Yes, I used `MissileDelay` for the missile, but it's for delaying the actual firing. So right now, opening sound is played when the building fires. Before that happens, you can see the animation without sound. But for the silo, it's quick enough, so you don't really notice it. But for PlasmaBlast, it will be more noticable, as there's bigger delay between starting the animation and firing, hence moving opening sound to `WithSupportPowerActivationAnimation` would be better.

**Comment by michaeldgg2** [comment]
`WithSupportPowerActivationAnimation` is executed when the support power is activated. And that happens in the same tick as when `LaunchSound` / `IncomingSound` (and other sounds such as speech notifications) are played. These are `UI`-type sounds, so if that is not enough, we need a new trait. `WithSupportPowerActivationAnimation` is very simple, but also focused when it comes to its function. Adding sound feature to it isn't really a good idea (it likely won't get past PR).

**Comment by Dzierzan** [comment]
Added PB Cannon and its superweapon.

plasmablast2

File: mods/e2140/content/ed/defenses/silo/weapons.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	Inherits: Smudges
+	Report: 130.smp
+	ValidTargets: Ground, Water, Ship
+	TargetActorCenter: true
+	Warhead@Damage: SpreadDamage
+		Spread: 1c156
+		Falloff: 100, 75, 45, 35, 5
+		Damage: 2000
+		DamageTypes: Explosion
+		ValidTargets: Ground, Ship
```
Comment (michaeldgg2) [inline mods/e2140/content/ed/defenses/silo/weapons.yaml]: Superweapons deal damage to airborne actors too

File: mods/e2140/content/ucs/defenses/pb_cannon/weapons.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+ucs_defenses_pb_cannon:
+	Report: 131.smp
+	ValidTargets: Ground, Water, Ship
+	TargetActorCenter: true
+	Warhead@Damage: SpreadDamage
+		Spread: 1c156
+		Falloff: 100, 75, 45, 35, 5
+		Damage: 1500
+		DamageTypes: Explosion
+		ValidTargets: Ground, Ship
```
Comment (michaeldgg2) [inline mods/e2140/content/ucs/defenses/pb_cannon/weapons.yaml]: Superweapons deal damage to airborne actors too

**Review by michaeldgg2 — CHANGES_REQUESTED** [review]

**Pull request merged**